code/service/src/main/scala/com/namely/chiefofstate/telemetry/TelemetryTools.scala

```diff
@@ -6,67 +6,93 @@
-  compositeRegistry.add(prometheusRegistry)
-  // create metrics reporter
-  val metricsReporter: MicrometerMetricsReporter = MicrometerMetricsReporter
-    .newMetricsReporter()
-    .withRegistry(compositeRegistry)
-    .withName(serviceName)
-    .withTagLabel(GRPC_STATUS_LABEL, "")
-    .build()
-
-  var tracer: Tracer = io.opentracing.noop.NoopTracerFactory.create()
-
-  if (enableJaeger) {
-    // create & register jaeger tracer
-    tracer = io.jaegertracing.Configuration.fromEnv().getTracer
+import io.opentelemetry.api.common.Attributes
+import io.opentelemetry.context.propagation.ContextPropagators
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter
+import io.opentelemetry.sdk.OpenTelemetrySdk
+import io.opentelemetry.sdk.metrics.SdkMeterProvider
+import io.opentelemetry.sdk.resources.Resource
+import io.opentelemetry.sdk.trace.SdkTracerProvider
+import io.opentelemetry.sdk.trace.`export`.BatchSpanProcessor
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes
+
+import java.util.Collections
+
+case class TelemetryTools(config: Config) {
+
+  def start(): Unit = {
+    val propagators: ContextPropagators = PropagatorConfiguration
+      .configurePropagators(config)
+
+    val resource = configureResource(config)
+
+    configureMetricsExporter(resource)
+
+    val tracerProvider = configureProvider(config, resource)
+
+    val sdkBuilder = OpenTelemetrySdk
+      .builder()
+      .setPropagators(propagators)
+
+    tracerProvider.map(sdkBuilder.setTracerProvider)
+
+    sdkBuilder.buildAndRegisterGlobal()
   }
 
-  // wrap tracer for metrics
-  tracer = io.opentracing.contrib.metrics.Metrics.decorate(tracer, metricsReporter)
-  // register tracer globally
-  GlobalTracer.registerIfAbsent(tracer)
-  // create prometheus scraping server
-  val prometheusServer: PrometheusServer = PrometheusServer(prometheusRegistry, config)
-
-  /**
-   * start all telemetry tools
-   *
-   * @return
-   */
-  def start(): TelemetryTools = {
-    logger.debug("starting telemetry tools")
-    this.prometheusServer.start()
-
-    sys.addShutdownHook {
-      stop()
+  def configureMetricsExporter(resource: Resource): Unit = {
+    val endpoint = config.getString("chiefofstate.otlp.endpoint")
+    if (!endpoint.isBlank) {
+      val meterProvider = SdkMeterProvider.builder.setResource(resource).buildAndRegisterGlobal
+      val builder = OtlpGrpcMetricExporter.builder()
+
+      builder.setEndpoint(endpoint)
+      val exporter = builder.build
+      val readerBuilder = IntervalMetricReader.builder
+        .setMetricProducers(Collections.singleton(meterProvider))
+        .setMetricExporter(exporter)
+      val reader = readerBuilder.build
+      sys.addShutdownHook {
+        reader.shutdown()
+      }
     }
+  }
+
+  def configureProvider(config: Config, resource: Resource): Option[SdkTracerProvider] = {
+    Option(config.getString("chiefofstate.otlp.endpoint"))
```

Comment:
Let us use the CosConfig to set this. That will help us know that all config are centralized.